pkgs/applications/misc/squeekboard/default.nix

```
@@ -0,0 +1,90 @@
+{ stdenv
+, pkgs
+, fetchFromGitLab
+, cmake
+, meson
+, ninja
+, pkg-config
+, gnome3
+, glib
+, gtk3
+, wayland
+, wayland-protocols
+, rustc
+, cargo
+, libxml2
+, libxkbcommon
+, rustPlatform
+, makeWrapper
+, substituteAll
+, fetchpatch
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "squeekboard";
+  version = "1.9.1";
+
+  src = fetchFromGitLab {
+    domain = "source.puri.sm";
+    owner = "Librem5";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1nrihpffr91p3ipif5k5z9vcvkwxk6qy8hp93w13s7gnwl2g93hx";
+  };
+
+  patches = [
+    ./desktop-in.patch
+    # This patch removes the unused dependency 'libcroco' the from meson.build
+    ( fetchpatch {
+      url = "https://source.puri.sm/Librem5/squeekboard/commit/f473a47eb8f394ab6f36704850e7e2bfa74ce8a1.patch";
+      sha256 = "0mlp8c38s4mbza8czf4kdg86kvqw294nbpqfk9apbl92nq0a26zr";
+    })
+  ];
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    rustc
+    cargo
+    glib.dev
+    gtk3.dev
```

Inputs just take all necessary outputs, no matter what you specify.

```suggestion
    glib
    gtk3
```

---

It would be also nice to have a comment why each of these is required since those are typically libraries, not build tools (e.g. `glib # for glib-mkenums` or `gtk3 # for gtk-update-icon-cache`).

---

I found glib isn't needed (is commented in meson.build)

About gtk3, looking at the code I think is used to render the keyboard (the buttons and other components)

---

should I add a comment on gtk3?

---

Is it used during build, or only linked against and used at run time. If the latter, it should go to regular `buildInputs`.

Also note that `gtk3` propagates `glib` so it might mask its necessity. You can try setting `strictDeps = true` to try to uncover it.

---

> Also note that gtk3 propagates glib so it might mask its necessity. You can try setting strictDeps = true to try to uncover it.

You were right. Fixed and added comments